great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -480,6 +491,31 @@ def get_domain_records(
 
         return data
 
+    def _combine_row_conditions(
+        self, row_conditions: List[RowCondition]
+    ) -> RowCondition:
+        """Combine row conditions using AND if type_ is SPARK_SQL
+
+        Args:
+            row_conditions: Row conditions of type Spark
+
+        Returns:
+            Single Row Condition combined
+        """
+        assert all(
+            [
+                condition.type_ == RowConditionParserType.SPARK_SQL
+                for condition in row_conditions
+            ]
+        ), "All row conditions must have type SPARK_SQL"
```

Comment:
Nitpick - could we get rid of the list here?

```python
all(condition.type == RowConditionParserType.SPARK_SQL for condition in row_conditions)
```

Does that work? Always get confused here but we can possibly skip creating a list.

Reply:
Yep! Good call - `all()` takes an iterable.